### PR TITLE
GC Timeout is Out of Date

### PR DESCRIPTION
**Pull request by Icemap**
### What is changed, added or deleted? (Required)

Update conflicting descriptions about GC in [`tidb_gc_life_time` system variable doc](https://docs.pingcap.com/zh/tidb/stable/system-variables#tidb_gc_life_time-%E4%BB%8E-v50-%E7%89%88%E6%9C%AC%E5%BC%80%E5%A7%8B%E5%BC%95%E5%85%A5) and [GC 超时 in dev guide](https://docs.pingcap.com/zh/tidb/stable/dev-guide-timeouts-in-tidb#gc-%E8%B6%85%E6%97%B6).

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [ ] v8.1 (TiDB 8.1 versions)
- [x] v8.0 (TiDB 8.0 versions)
- [x] v7.6 (TiDB 7.6 versions)
- [x] v7.5 (TiDB 7.5 versions)
- [ ] v7.4 (TiDB 7.4 versions)
- [x] v7.1 (TiDB 7.1 versions)
- [x] v6.5 (TiDB 6.5 versions)
- [x] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: develop/dev-guide-timeouts-in-tidb.md [hunk 1]
```diff
@@ -12,9 +12,15 @@ aliases: ['/zh/tidb/dev/timeouts-in-tidb']
 
 TiDB 的事务的实现采用了 MVCC（多版本并发控制）机制，当新写入的数据覆盖旧的数据时，旧的数据不会被替换掉，而是与新写入的数据同时保留，并以时间戳来区分版本。TiDB 通过定期 GC 的机制来清理不再需要的旧数据。
 
-默认配置下 TiDB 可以保障每个 MVCC 版本（一致性快照）保存 10 分钟，读取时间超过 10 分钟的事务，会收到报错 `GC life time is shorter than transaction duration`。
+- 使用 4.0 之前版本的 TiDB：
 
-当用户确信自己需要更长的读取时间时，比如在使用了 Dumpling 做全量备份的场景中（当 Dumpling 备份的是一致性的快照），可以通过调整 TiDB 中`mysql.tidb` 表中的 `tikv_gc_life_time` 的值来调大 MVCC 版本保留时间，需要注意的是 tikv_gc_life_time 的配置是立刻影响全局的，调大它会为当前所有存在的快照增加生命时长，调小它会立即缩短所有快照的生命时长。过多的 MVCC 版本会拖慢 TiKV 的处理效率，在使用 Dumpling 做完全量备份后需要及时把 `tikv_gc_life_time` 调整回之前的设置。
+    默认配置下 TiDB 可以保障每个 MVCC 版本（一致性快照）保存 10 分钟，读取时间超过 10 分钟的事务，会收到报错 `GC life time is shorter than transaction duration`。
+
+- 使用 4.0 及更新版本的 TiDB：
```
Comment (lilin90) [inline develop/dev-guide-timeouts-in-tidb.md:19]:
```suggestion
- 使用 4.0 及之后版本的 TiDB：
```

File: develop/dev-guide-timeouts-in-tidb.md [hunk 2]
```diff
@@ -12,9 +12,15 @@ aliases: ['/zh/tidb/dev/timeouts-in-tidb']
...
+
+    正在运行的事务，如果持续时间不超过 24 小时，在运行期间 GC 会被阻塞，不会出现 `GC life time is shorter than transaction duration` 报错。
+
+当用户确信自己需要更长的读取时间时，比如在使用了 Dumpling 做全量备份的场景中（当 Dumpling 备份的是一致性的快照），可以通过调整 TiDB 中 `mysql.tidb` 表中的 `tikv_gc_life_time` 的值来调大 MVCC 版本保留时间，需要注意的是 tikv_gc_life_time 的配置是立刻影响全局的，调大它会为当前所有存在的快照增加生命时长，调小它会立即缩短所有快照的生命时长。过多的 MVCC 版本会拖慢 TiKV 的处理效率，在使用 Dumpling 做完全量备份后需要及时把 `tikv_gc_life_time` 调整回之前的设置。
```
Comment (lilin90) [inline develop/dev-guide-timeouts-in-tidb.md:23]:
```suggestion
当用户确信自己需要更长的读取时间时，比如在使用了 Dumpling 做全量备份的场景中（当 Dumpling 备份的是一致性的快照），可以通过调整 TiDB 中 `mysql.tidb` 表中的 `tikv_gc_life_time` 的值来调大 MVCC 版本保留时间，需要注意的是 `tikv_gc_life_time` 的配置是立刻影响全局的，调大它会为当前所有存在的快照增加生命时长，调小它会立即缩短所有快照的生命时长。过多的 MVCC 版本会拖慢 TiKV 的处理效率，在使用 Dumpling 做完全量备份后需要及时把 `tikv_gc_life_time` 调整回之前的设置。
```

File: develop/dev-guide-timeouts-in-tidb.md [hunk 3]
```diff
@@ -12,9 +12,15 @@ aliases: ['/zh/tidb/dev/timeouts-in-tidb']
...
+- 使用 4.0 之前版本的 TiDB：
 
-当用户确信自己需要更长的读取时间时，比如在使用了 Dumpling 做全量备份的场景中（当 Dumpling 备份的是一致性的快照），可以通过调整 TiDB 中`mysql.tidb` 表中的 `tikv_gc_life_time` 的值来调大 MVCC 版本保留时间，需要注意的是 tikv_gc_life_time 的配置是立刻影响全局的，调大它会为当前所有存在的快照增加生命时长，调小它会立即缩短所有快照的生命时长。过多的 MVCC 版本会拖慢 TiKV 的处理效率，在使用 Dumpling 做完全量备份后需要及时把 `tikv_gc_life_time` 调整回之前的设置。
+    默认配置下 TiDB 可以保障每个 MVCC 版本（一致性快照）保存 10 分钟，读取时间超过 10 分钟的事务，会收到报错 `GC life time is shorter than transaction duration`。
+
+- 使用 4.0 及之后版本的 TiDB：
+
+    正在运行的事务，如果持续时间不超过 24 小时，在运行期间 GC 会被阻塞，不会出现 `GC life time is shorter than transaction duration` 报错。
+
+当用户确信自己需要更长的读取时间时，比如在使用了 Dumpling 做全量备份的场景中（当 Dumpling 备份的是一致性的快照），可以通过调整 TiDB 中 `mysql.tidb` 表中的 `tikv_gc_life_time` 的值来调大 MVCC 版本保留时间，需要注意的是 `tikv_gc_life_time` 的配置是立刻影响全局的，调大它会为当前所有存在的快照增加生命时长，调小它会立即缩短所有快照的生命时长。过多的 MVCC 版本会拖慢 TiKV 的处理效率，在使用 Dumpling 做完全量备份后需要及时把 `tikv_gc_life_time` 调整回之前的设置。
```
Comment (MyonKeminta) [inline develop/dev-guide-timeouts-in-tidb.md:23]: From v5.0 the system variable `tidb_gc_life_time` is introduced, while directly updating `mysql.tidb` table becomes unrecommended. Please consider update it to using system variable `tidb_gc_life_time`, and better add a hyperlink to the document of the variable.

Comment (Icemap) [reply develop/dev-guide-timeouts-in-tidb.md]: Thanks for advising. I changed it to describe them separately. PTAL.

File: develop/dev-guide-timeouts-in-tidb.md [hunk 4]
```diff
@@ -12,9 +12,36 @@ aliases: ['/zh/tidb/dev/timeouts-in-tidb']
...
+
+- 使用 5.0 及之后版本的 TiDB：
+
+    系统变量 [`tidb_gc_life_time`](/system-variables.md#tidb_gc_life_time-从-v50-版本开始引入)
+
+来调大 MVCC 版本保留时间，需要注意的是此变量的配置是立刻影响全局的，调大它会为当前所有存在的快照增加生命时长，调小它会立即缩短所有快照的生命时长。过多的 MVCC 版本会影响 TiDB 的集群性能，在使用后需要及时把此变量调整回之前的设置。
+
+> **Tip:**
+>
+> 特别地，在 Dumpling 备份时，如果导出的数据量少于 1 TB 且导出的 TiDB 版本为 v4.0.0 或更新版本，并且 Dumpling 可以访问 TiDB 集群的 PD 地址，Dumpling 会自动配置延长 GC 时间且不会对原集群造成影响。以下场景除外：
```
Comment (lilin90) [inline develop/dev-guide-timeouts-in-tidb.md:37]:
```suggestion
> 特别地，在 Dumpling 备份时，如果导出的数据量少于 1 TiB 且导出的 TiDB 版本为 v4.0.0 或更新版本，并且 Dumpling 可以访问 TiDB 集群的 PD 地址，Dumpling 会自动配置延长 GC 时间且不会对原集群造成影响。以下场景除外：
```

File: develop/dev-guide-timeouts-in-tidb.md [hunk 5]
```diff
@@ -12,9 +12,36 @@ aliases: ['/zh/tidb/dev/timeouts-in-tidb']
...
+>
+> - 数据量非常大（超过 1 TB）。
```
Comment (lilin90) [inline develop/dev-guide-timeouts-in-tidb.md:39]:
```suggestion
> - 数据量非常大（超过 1 TiB）。
```

File: develop/dev-guide-timeouts-in-tidb.md [hunk 6]
```diff
@@ -12,9 +12,36 @@ aliases: ['/zh/tidb/dev/timeouts-in-tidb']
...
+
+- 使用 5.0 之前版本的 TiDB：
+
+    `mysql.tidb` 表中的 `tikv_gc_life_time`
+
+- 使用 5.0 及之后版本的 TiDB：
+
+    系统变量 [`tidb_gc_life_time`](/system-variables.md#tidb_gc_life_time-从-v50-版本开始引入)
+
+来调大 MVCC 版本保留时间，需要注意的是此变量的配置是立刻影响全局的，调大它会为当前所有存在的快照增加生命时长，调小它会立即缩短所有快照的生命时长。过多的 MVCC 版本会影响 TiDB 的集群性能，在使用后需要及时把此变量调整回之前的设置。
```
Comment (lilin90) [inline develop/dev-guide-timeouts-in-tidb.md:33]:
```suggestion
当用户确信自己在临时特殊场景中需要更长的读取时间时，可以通过以下方式调大 MVCC 版本保留时间：

- 使用 5.0 之前版本的 TiDB：调整 `mysql.tidb` 表中的 `tikv_gc_life_time`。

- 使用 5.0 及之后版本的 TiDB：调整系统变量 [`tidb_gc_life_time`](/system-variables.md#tidb_gc_life_time-从-v50-版本开始引入)。

需要注意的是，此变量的配置是立刻影响全局的，调大它会为当前所有存在的快照增加生命时长，调小它会立即缩短所有快照的生命时长。过多的 MVCC 版本会影响 TiDB 的集群性能，在使用后需要及时把此变量调整回之前的设置。
```

File: develop/dev-guide-timeouts-in-tidb.md (hunk 4 above)
Comment (Icemap) [reply develop/dev-guide-timeouts-in-tidb.md:37]: @lilin90 这里是和 https://docs.pingcap.com/zh/tidb/stable/dumpling-overview#%E6%89%8B%E5%8A%A8%E8%AE%BE%E7%BD%AE-tidb-gc-%E6%97%B6%E9%97%B4 保持一致的，是否需要统一更改？

Comment (lilin90) [reply develop/dev-guide-timeouts-in-tidb.md:37]: > @lilin90 这里是和 https://docs.pingcap.com/zh/tidb/stable/dumpling-overview#%E6%89%8B%E5%8A%A8%E8%AE%BE%E7%BD%AE-tidb-gc-%E6%97%B6%E9%97%B4 保持一致的，是否需要统一更改？

那先保持现状吧，统一改涉及不少文档，之后再另提 PR。上个月在 https://github.com/pingcap/docs-cn/pull/16844 里专门改过系统变量和 TiKV 配置文件里的。

Comment (Icemap) [reply develop/dev-guide-timeouts-in-tidb.md:37]: 好的。

File: develop/dev-guide-timeouts-in-tidb.md [hunk 7]
```diff
@@ -12,9 +12,32 @@ aliases: ['/zh/tidb/dev/timeouts-in-tidb']
...
+    默认配置下 TiDB 可以保障每个 MVCC 版本（一致性快照）保存 10 分钟，读取时间超过 10 分钟的事务，会收到报错 `GC life time is shorter than transaction duration`。
+
+- 使用 4.0 及之后版本的 TiDB：
+
+    正在运行的事务，如果持续时间不超过 24 小时，在运行期间 GC 会被阻塞，不会出现 `GC life time is shorter than transaction duration` 报错。
+
+当用户确信自己在临时特殊场景中需要更长的读取时间时，可以通过以下方式调大 MVCC 版本保留时间：
+
+- 使用 5.0 之前版本的 TiDB：调整 `mysql.tidb` 表中的 `tikv_gc_life_time`。
+
```
Comment (Icemap) [inline develop/dev-guide-timeouts-in-tidb.md]:
```suggestion
```

**Comment by qiancai** [comment]
Removed the needs-cherry-pick-release-7.4 label because the v7.4 docs have been archived at https://docs-archive.pingcap.com/zh/tidb/v7.4 and will no longer receive new updates.

**Comment by lilin90** [comment]
Note: The `release-8.1` branch hasn't been created, so I removed the corresponding cherry pick label.

File: develop/dev-guide-timeouts-in-tidb.md [hunk 8]
```diff
@@ -12,9 +12,31 @@ aliases: ['/zh/tidb/dev/timeouts-in-tidb']
...
+- 使用 4.0 之前版本的 TiDB：
 
-当用户确信自己需要更长的读取时间时，比如在使用了 Dumpling 做全量备份的场景中（当 Dumpling 备份的是一致性的快照），可以通过调整 TiDB 中`mysql.tidb` 表中的 `tikv_gc_life_time` 的值来调大 MVCC 版本保留时间，需要注意的是 tikv_gc_life_time 的配置是立刻影响全局的，调大它会为当前所有存在的快照增加生命时长，调小它会立即缩短所有快照的生命时长。过多的 MVCC 版本会拖慢 TiKV 的处理效率，在使用 Dumpling 做完全量备份后需要及时把 `tikv_gc_life_time` 调整回之前的设置。
+    默认配置下 TiDB 可以保障每个 MVCC 版本（一致性快照）保存 10 分钟，读取时间超过 10 分钟的事务，会收到报错 `GC life time is shorter than transaction duration`。
+
+- 使用 4.0 及之后版本的 TiDB：
+
+    正在运行的事务，如果持续时间不超过 24 小时，在运行期间 GC 会被阻塞，不会出现 `GC life time is shorter than transaction duration` 报错。
+
+当用户确信自己在临时特殊场景中需要更长的读取时间时，可以通过以下方式调大 MVCC 版本保留时间：
```
Comment (lilin90) [inline develop/dev-guide-timeouts-in-tidb.md:23]: 与下文的第二人称表述保持一致。
```suggestion
当你确信自己在临时特殊场景中需要更长的读取时间时，可以通过以下方式调大 MVCC 版本保留时间：
```

File: develop/dev-guide-timeouts-in-tidb.md [hunk 9]
```diff
@@ -12,9 +12,31 @@ aliases: ['/zh/tidb/dev/timeouts-in-tidb']
...
+> **Tip:**
+>
+> 特别地，在 Dumpling 备份时，如果导出的数据量少于 1 TB 且导出的 TiDB 版本为 v4.0.0 或更新版本，并且 Dumpling 可以访问 TiDB 集群的 PD 地址，Dumpling 会自动配置延长 GC 时间且不会对原集群造成影响。以下场景除外：
+>
+> - 数据量非常大（超过 1 TB）。
+> - Dumpling 无法直接连接到 PD，例如 TiDB 集群运行在 TiDB Cloud 上，或者 TiDB 集群运行在 Kubernetes 上且与 Dumpling 分离。
+>
+> 在这些场景中，你必须使用 `tikv_gc_life_time` 提前手动调长 GC 时间，以避免因为导出过程中发生 GC 导致导出失败。
+>
+> 详情请见 TiDB 工具 Dumpling 的[手动设置 TiDB GC 时间](/dumpling-overview.md#手动设置-tidb-gc-时间)一节。
```
Comment (lilin90) [inline develop/dev-guide-timeouts-in-tidb.md:39]:
```suggestion
> 在这些场景中，你必须使用 `tikv_gc_life_time` 提前手动调长 GC 时间，以避免因为导出过程中发生 GC 导致导出失败。详见 TiDB 工具 Dumpling 的[手动设置 TiDB GC 时间](/dumpling-overview.md#手动设置-tidb-gc-时间)。
```

**Review by lilin90 — APPROVED** [review]
/lgtm

**Review by MyonKeminta — APPROVED** [review]

File: develop/dev-guide-timeouts-in-tidb.md [hunk 10]
```diff
@@ -12,9 +12,29 @@ aliases: ['/zh/tidb/dev/timeouts-in-tidb']
 
 TiDB 的事务的实现采用了 MVCC（多版本并发控制）机制，当新写入的数据覆盖旧的数据时，旧的数据不会被替换掉，而是与新写入的数据同时保留，并以时间戳来区分版本。TiDB 通过定期 GC 的机制来清理不再需要的旧数据。
 
-默认配置下 TiDB 可以保障每个 MVCC 版本（一致性快照）保存 10 分钟，读取时间超过 10 分钟的事务，会收到报错 `GC life time is shorter than transaction duration`。
+- 使用 4.0 之前版本的 TiDB：
 
-当用户确信自己需要更长的读取时间时，比如在使用了 Dumpling 做全量备份的场景中（当 Dumpling 备份的是一致性的快照），可以通过调整 TiDB 中`mysql.tidb` 表中的 `tikv_gc_life_time` 的值来调大 MVCC 版本保留时间，需要注意的是 tikv_gc_life_time 的配置是立刻影响全局的，调大它会为当前所有存在的快照增加生命时长，调小它会立即缩短所有快照的生命时长。过多的 MVCC 版本会拖慢 TiKV 的处理效率，在使用 Dumpling 做完全量备份后需要及时把 `tikv_gc_life_time` 调整回之前的设置。
+    默认配置下 TiDB 可以保障每个 MVCC 版本（一致性快照）保存 10 分钟，读取时间超过 10 分钟的事务，会收到报错 `GC life time is shorter than transaction duration`。
```
Comment (hfxsd) [inline develop/dev-guide-timeouts-in-tidb.md:17]:
```suggestion
    默认情况下，TiDB 可以确保每个 MVCC 版本（一致性快照）保存 10 分钟。读取时间超过 10 分钟的事务，会收到报错 `GC life time is shorter than transaction duration`。
```

File: develop/dev-guide-timeouts-in-tidb.md [hunk 11]
```diff
@@ -12,9 +12,29 @@ aliases: ['/zh/tidb/dev/timeouts-in-tidb']
 
 TiDB 的事务的实现采用了 MVCC（多版本并发控制）机制，当新写入的数据覆盖旧的数据时，旧的数据不会被替换掉，而是与新写入的数据同时保留，并以时间戳来区分版本。TiDB 通过定期 GC 的机制来清理不再需要的旧数据。
 
-默认配置下 TiDB 可以保障每个 MVCC 版本（一致性快照）保存 10 分钟，读取时间超过 10 分钟的事务，会收到报错 `GC life time is shorter than transaction duration`。
+- 使用 4.0 之前版本的 TiDB：
```
Comment (hfxsd) [inline develop/dev-guide-timeouts-in-tidb.md:15]:
```suggestion
- TiDB v4.0 之前的版本：
```

File: develop/dev-guide-timeouts-in-tidb.md [hunk 12]
```diff
@@ -12,9 +12,29 @@ aliases: ['/zh/tidb/dev/timeouts-in-tidb']
...
 
-当用户确信自己需要更长的读取时间时，比如在使用了 Dumpling 做全量备份的场景中（当 Dumpling 备份的是一致性的快照），可以通过调整 TiDB 中`mysql.tidb` 表中的 `tikv_gc_life_time` 的值来调大 MVCC 版本保留时间，需要注意的是 tikv_gc_life_time 的配置是立刻影响全局的，调大它会为当前所有存在的快照增加生命时长，调小它会立即缩短所有快照的生命时长。过多的 MVCC 版本会拖慢 TiKV 的处理效率，在使用 Dumpling 做完全量备份后需要及时把 `tikv_gc_life_time` 调整回之前的设置。
+    默认配置下 TiDB 可以保障每个 MVCC 版本（一致性快照）保存 10 分钟，读取时间超过 10 分钟的事务，会收到报错 `GC life time is shorter than transaction duration`。
+
+- 使用 4.0 及之后版本的 TiDB：
```
Comment (hfxsd) [inline develop/dev-guide-timeouts-in-tidb.md:19]:
```suggestion
- TiDB v4.0 及之后的版本：
```

File: develop/dev-guide-timeouts-in-tidb.md [hunk 13]
```diff
@@ -12,9 +12,29 @@ aliases: ['/zh/tidb/dev/timeouts-in-tidb']
...
+
+    正在运行的事务，如果持续时间不超过 24 小时，在运行期间 GC 会被阻塞，不会出现 `GC life time is shorter than transaction duration` 报错。
+
+当你确信自己在临时特殊场景中需要更长的读取时间时，可以通过以下方式调大 MVCC 版本保留时间：
```
Comment (hfxsd) [inline develop/dev-guide-timeouts-in-tidb.md:23]:
```suggestion
如果你确定在临时特殊场景中需要更长的读取时间，可以通过以下方式调大 MVCC 版本保留时间：
```

File: develop/dev-guide-timeouts-in-tidb.md [hunk 14]
```diff
@@ -12,9 +12,29 @@ aliases: ['/zh/tidb/dev/timeouts-in-tidb']
...
+
+- 使用 5.0 之前版本的 TiDB：调整 `mysql.tidb` 表中的 `tikv_gc_life_time`。
+- 使用 5.0 及之后版本的 TiDB：调整系统变量 [`tidb_gc_life_time`](/system-variables.md#tidb_gc_life_time-从-v50-版本开始引入)。
```
Comment (hfxsd) [inline develop/dev-guide-timeouts-in-tidb.md:26]:
```suggestion
- TiDB v5.0 之前的版本 ：调整 `mysql.tidb` 表中的 `tikv_gc_life_time`。
- TiDB v5.0 及之后的版本：调整系统变量 [`tidb_gc_life_time`](/system-variables.md#tidb_gc_life_time-从-v50-版本开始引入)。
```

File: develop/dev-guide-timeouts-in-tidb.md [hunk 15]
```diff
@@ -12,9 +12,29 @@ aliases: ['/zh/tidb/dev/timeouts-in-tidb']
...
+
+需要注意的是，此变量的配置是立刻影响全局的，调大它会为当前所有存在的快照增加生命时长，调小它会立即缩短所有快照的生命时长。过多的 MVCC 版本会影响 TiDB 的集群性能，在使用后需要及时把此变量调整回之前的设置。
```
Comment (hfxsd) [inline develop/dev-guide-timeouts-in-tidb.md:28]:
```suggestion
需要注意的是，此变量的配置是立刻影响全局的，调大它会增加当前所有快照的生命时长，调小它也会立即缩短所有快照的生命时长。过多的 MVCC 版本会影响 TiDB 的集群性能，因此在使用后，需要及时把此变量调整回之前的设置。
```

**Review by hfxsd — APPROVED** [review]

**Pull request merged**